src/Build.UnitTests/BackEnd/TargetBuilder_Tests.cs

```diff
@@ -1484,6 +1538,30 @@ private ProjectLoggingContext GetProjectLoggingContext(BuildRequestEntry entry)
             return new ProjectLoggingContext(new NodeLoggingContext(_host, 1, false), entry, null);
         }
 
+        /// <summary>
+        /// Builds a project using TargetBuilder and returns the result.
+        /// </summary>
+        /// <param name="projectBody">The project contents.</param>
+        /// <param name="targets">The targets to build.</param>
+        /// <param name="failTaskNumber">The task ordinal to fail on, or null if no task should fail.</param>
+        /// <returns>The result of building the specified project/tasks.</returns>
+        private BuildResult BuildSimpleProject(string projectBody, string[] targets, int? failTaskNumber = null)
+        {
+            ProjectInstance project = CreateTestProject(projectBody);
+
+            if (failTaskNumber.HasValue)
```

Comment:
Does it ever not have a value? Could you just make it non-nullable?

Reply:
I wanted to encourage reuse of this method by tests that do not require a task failure. Fixed as you're suggesting to avoid confusion.